platform/ios/src/MGLMapView.mm

```diff
@@ -1061,29 +1056,56 @@ - (void)installAttributionButtonConstraints {
     [self updateConstraintsForOrnament:self.attributionButton
                            constraints:self.attributionButtonConstraints
                               position:self.attributionButtonPosition
-                                  size:[self sizeForOrnament:self.attributionButton constraints:self.attributionButtonConstraints]
+                       sizeConstraints:[self sizeConstraintsForOrnament:self.attributionButton constraints:self.attributionButtonConstraints]
                                margins:self.attributionButtonMargins];
 }
 
-- (CGSize)sizeForOrnament:(UIView *)view
-              constraints:(NSMutableArray *)constraints {
-    // avoid regenerating size constraints
-    CGSize size;
-    if(constraints && constraints.count > 0) {
-        for (NSLayoutConstraint * constraint in constraints) {
-            if([constraint.identifier isEqualToString:@"width"]) {
-                size.width = constraint.constant;
-            }
-            else if ([constraint.identifier isEqualToString:@"height"]) {
-                size.height = constraint.constant;
-            }
-        }
+- (NSArray *)sizeConstraintsForOrnament:(UIView *)view
+                            constraints:(NSMutableArray *)constraints {
+
+    NSLayoutConstraint *widthConstraint = nil;
+    NSLayoutConstraint *heightConstraint = nil;
+
+    if (view == _scaleBar) {
+        widthConstraint = [NSLayoutConstraint constraintWithItem:_scaleBar
+                                                       attribute:NSLayoutAttributeWidth
+                                                       relatedBy:NSLayoutRelationEqual
+                                                          toItem:self
+                                                       attribute:NSLayoutAttributeWidth
+                                                      multiplier:.5
+                                                        constant:0];
+        heightConstraint = [NSLayoutConstraint constraintWithItem:_scaleBar
+                                                        attribute:NSLayoutAttributeHeight
+                                                        relatedBy:NSLayoutRelationEqual
+                                                           toItem:nil
+                                                        attribute:NSLayoutAttributeNotAnAttribute
+                                                       multiplier:1
+                                                         constant:16];
```

Comment:
Do we have an existing constant for this?

Reply:
Yes, I get the height value from https://github.com/mapbox/mapbox-gl-native-ios/pull/507/files/ab5ff48099caff30598e35b2d7a2edc49539babd#diff-6f8e414d71df96dda5dfecbd275ee6c5a35b52e337f3f941c6c132abda50ffb0L347